sdk/python/feast/infra/utils/snowflake_utils.py

```diff
@@ -429,3 +433,176 @@ def parse_private_key_path(key_path: str, private_key_passphrase: str) -> bytes:
+    compression_map = {"gzip": "auto", "snappy": "snappy"}
+    if compression not in compression_map.keys():
+        raise ProgrammingError(
+            "Invalid compression '{}', only acceptable values are: {}".format(
+                compression, compression_map.keys()
+            )
+        )
+    if quote_identifiers:
+        location = (
+            (('"' + database + '".') if database else "")
+            + (('"' + schema + '".') if schema else "")
+            + ('"' + table_name + '"')
+        )
+    else:
+        location = (
+            (database + "." if database else "")
+            + (schema + "." if schema else "")
+            + (table_name)
+        )
+    if chunk_size is None:
+        chunk_size = len(df)
+    cursor: SnowflakeCursor = conn.cursor()
+    stage_name = create_temporary_sfc_stage(cursor)
+
+    with TemporaryDirectory() as tmp_folder:
+        for i, chunk in chunk_helper(df, chunk_size):
+            chunk_path = os.path.join(tmp_folder, "file{}.txt".format(i))
+            # Dump chunk into parquet file
+            chunk.to_parquet(
+                chunk_path,
+                compression=compression,
+                use_deprecated_int96_timestamps=True,
+            )
+            # Upload parquet file
+            upload_sql = (
+                "PUT /* Python:snowflake.connector.pandas_tools.write_pandas() */ "
+                "'file://{path}' @\"{stage_name}\" PARALLEL={parallel}"
+            ).format(
+                path=chunk_path.replace("\\", "\\\\").replace("'", "\\'"),
+                stage_name=stage_name,
+                parallel=parallel,
+            )
+            logger.debug(f"uploading files with '{upload_sql}'")
+            cursor.execute(upload_sql, _is_internal=True)
+            # Remove chunk file
+            os.remove(chunk_path)
+    if quote_identifiers:
+        columns = '"' + '","'.join(list(df.columns)) + '"'
+    else:
+        columns = ",".join(list(df.columns))
+
+    if auto_create_table:
+        file_format_name = create_file_format(compression, compression_map, cursor)
+        infer_schema_sql = f"SELECT COLUMN_NAME, TYPE FROM table(infer_schema(location=>'@\"{stage_name}\"', file_format=>'{file_format_name}'))"
+        logger.debug(f"inferring schema with '{infer_schema_sql}'")
+        result_cursor = cursor.execute(infer_schema_sql, _is_internal=True)
+        if result_cursor is None:
+            raise SnowflakeQueryUnknownError(infer_schema_sql)
+        result = cast(List[Tuple[str, str]], result_cursor.fetchall())
+        column_type_mapping: Dict[str, str] = dict(result)
+        # Infer schema can return the columns out of order depending on the chunking we do when uploading
+        # so we have to iterate through the dataframe columns to make sure we create the table with its
+        # columns in order
+        quote = '"' if quote_identifiers else ""
```

Comment:
this can probably also be used earlier for consistency

Reply:
This is related specifically to quoting newly created table names that doesnt apply if auto_create_table=False